internal/manifests/servicemonitor/servicemonitor.go

```diff
@@ -0,0 +1,144 @@
+	tempo := params.Tempo
+	scheme := "http"
+	var tlsConfig *monitoringv1.TLSConfig
+
+	if params.Gates.HTTPEncryption {
+		scheme = "https"
+		serviceName := naming.ServiceName(tempo.Name, component, manifestutils.HttpPortName)
+		serverName := naming.ServiceFqdn(tempo.Namespace, tempo.Name, component, manifestutils.HttpPortName)
+
+		tlsConfig = &monitoringv1.TLSConfig{
+			SafeTLSConfig: monitoringv1.SafeTLSConfig{
+				CA: monitoringv1.SecretOrConfigMap{
+					ConfigMap: &corev1.ConfigMapKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: naming.SigningCABundleName(tempo.Name),
+						},
+						Key: certrotation.CAFile,
+					},
+				},
+				Cert: monitoringv1.SecretOrConfigMap{
+					Secret: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: serviceName,
+						},
+						Key: corev1.TLSCertKey,
+					},
+				},
+				KeySecret: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: serviceName,
+					},
+					Key: corev1.TLSPrivateKeyKey,
+				},
+				// E.g. tempo-simplest-compactor-http.tempo-operator.svc.cluster.local
+				ServerName: serverName,
+			},
+		}
+	}
+
+	return &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: tempo.Namespace,
+			Name:      naming.Name(component, tempo.Name),
+			Labels:    manifestutils.CommonLabels(tempo.Name),
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{{
+				Scheme:    scheme,
+				Port:      manifestutils.HttpPortName,
+				Path:      "/metrics",
+				TLSConfig: tlsConfig,
+				// Custom relabel configs to be compatible with predefined Tempo dashboards:
+				// https://grafana.com/docs/tempo/latest/operations/monitoring/#dashboards
+				RelabelConfigs: []*monitoringv1.RelabelConfig{
+					{
+						SourceLabels: []monitoringv1.LabelName{"__meta_kubernetes_service_label_app_kubernetes_io_instance"},
+						TargetLabel:  "cluster",
+					},
+					{
+						SourceLabels: []monitoringv1.LabelName{"__meta_kubernetes_namespace", "__meta_kubernetes_service_label_app_kubernetes_io_component"},
+						Separator:    "/",
+						TargetLabel:  "job",
+					},
+				},
+			}},
+			NamespaceSelector: monitoringv1.NamespaceSelector{
+				MatchNames: []string{tempo.Namespace},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: manifestutils.ComponentLabels(component, tempo.Name),
+			},
+		},
+	}
+}
+
+func buildGatewayServiceMonitor(tempo v1alpha1.TempoStack) *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: tempo.Namespace,
+			Name:      naming.Name(manifestutils.GatewayComponentName, tempo.Name),
+			Labels:    manifestutils.CommonLabels(tempo.Name),
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{{
+				Port: gateway.InternalPortName,
```

Comment:
@frzifus is this the proper port to get metrics from the gateway itself? It returns a few default metrics.
Also, does the metrics endpoint support TLS?

Reply:
good question, i would expect yes. But ive to look it up in the code or check if its configured in RHOBS.